pkg/constants/constants.go

```diff
@@ -0,0 +1,14 @@
+package constants
+
+import (
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1alpha1"
+)
+
+const (
+	mergedPullSecretSuffix = "merged-pull-secret"
+)
+
+// GetMergedPullSecretName returns name for merged pull secret name per cluster deployment
+func GetMergedPullSecretName(cd *hivev1.ClusterDeployment) string {
+	return cd.Namespace + "-" + mergedPullSecretSuffix
+}
```

Comment:
This should be cd.Name not cd.Namespace, as it would still generate the same secret name for two clusters in the same ns.

Once switching to cd.Name, it still looks like it can exceed the length limits for kubernetes object names with a long cluster deployment name, which is a problem that has actually surfaced in real usage. See my comment below on how to mitigate this with the namer.go GetName function.

Reply:
+1 to cd.Name. Hopefully `apihelpers.GetResourceName` returns same name for same argument every time else we will have failures as the merged secret name is used in different places.